SDWebImage/UIImage+Transform.m

```diff
@@ -349,7 +349,7 @@ - (UIImage *)sd_tintedImageWithColor:(UIColor *)tintColor {
     
     SDGraphicsBeginImageContextWithOptions(size, NO, scale);
     CGContextRef context = SDGraphicsGetCurrentContext();
-    CGContextDrawImage(context, rect, self.CGImage);
+    [self drawInRect:rect];
     CGContextSetBlendMode(context, blendMode);
```

Comment:
this method used to work with CoreGraphics functions only. I would be curious to know what other change in the lib has caused this upside down issue.
I'm not saying it's a bad fix, but it sure looks like it doesn't match the rest of the method body

Reply:
Actually, this code can be placed like this:

```objective-c
#if SD_UIKIT || SD_WATCH
CGContextScaleCTM(context, 1.0, -1.0);
CGContextTranslateCTM(context, 0, -size.height);
#endif 
CGContextDrawImage(context, rect, self.CGImage);
```

But using that `drawInRect:` seems a little convienet than Core Graphics APIs. So I think it's OK.

Reply:
@bpoplauschi We update all test cases to ensure this does not cause bug in the future. Currently only this and `sd_roundedCornerImageWithRadius:` contains the same issue.

Reply:
got it. thanks @dreampiggy